flang/include/flang/Optimizer/HLFIR/HLFIROps.td

```diff
@@ -404,6 +404,32 @@ def hlfir_CountOp : hlfir_Op<"count", [AttrSizedOperandSegments, DeclareOpInterf
   let hasVerifier = 1;
 }
 
+def hlfir_MaxvalOp : hlfir_Op<"maxval", [AttrSizedOperandSegments,
+    DeclareOpInterfaceMethods<ArithFastMathInterface>]> {
```

Comment:
Please could you add memory effects like I did in https://reviews.llvm.org/D158662. This is new since the previous intrinsics were added.

You will need to `DeclareOpInterfaceMethods<MemoryEffectsOpInterface>`, add an implementation of `MaxValOp::getEffects` which calls `getIntrinsicEffects` and add a test to `memory-effects.fir`

Reply:
Thank you for your comment and information.
I fixed it.